Semi realtime midi - simplify rhythms

shoogle opened the pull request:
I have prepared some test cases. I'll add them later today once I have double-checked them.

lasconic commented on libmscore/chord.cpp:
```diff
@@ -3098,7 +3098,36 @@ void Chord::sortNotes()
+      Chord* next = toChord(nextCR);
+      if (sameSize && notes().size() != next->notes().size())
+            return 0; // sizes don't match so some notes can't be tied
+      for (auto n = notes().begin(); n != notes().end(); ++n) {
```
`for (Note* n : _notes) {`?

lasconic commented on libmscore/chord.cpp:
```diff
@@ -3098,7 +3098,36 @@ void Chord::sortNotes()
+//    back to this one. Set sameSize=true to return 0 in this case.
+//---------------------------------------------------------
+
+Chord* Chord::nextTiedChord(bool backwards, bool sameSize)
```
What's exactly the purpose of `backward`

shoogle replied:
It was supposed to mean "look backwards" (i.e. return the previous one instead of the next one). It worked when I was using `segment()->nextChordRest(track, backwards);`, but then I swapped that for `segment()->next1(Segment::Type::ChordRest);` and it looks like I forgot to handle the backwards case.

lasconic replied:
OK. when fixed, please add a comment.

shoogle replied:
Done. It doesn't show up in the preview on the comment page for some reason, but [it's there](https://github.com/musescore/MuseScore/pull/2784/files#diff-428e0bb12d899d2d494b3cc99e75e22fR3106).

lasconic commented:
Tests are failing and one bug report.

Reset tie and rest for the following. It doesn't work for voice 2. On first voice, setNoteRest is called, and set the selection to single. Next call, selection is single and so start/endtick are not good. A simple fix would be to keep start/endtick in a local variable. [MSCZ](https://dl.dropboxusercontent.com/u/497455/multivoice_test.mscz) 

<img width="553" alt="capture d ecran 2016-08-15 12 33 02" src="https://cloud.githubusercontent.com/assets/234271/17662269/a3396b56-62e4-11e6-927e-3c00865ae730.png">

shoogle commented:
Still work to do on this, but I fixed the tests and added new ones. Have a look at the [reference files](https://github.com/shoogle/MuseScore/tree/d7a62a7dce3c23825599cae96c4cc9012b7e1564/mtest/libmscore/rhythmicGrouping) to see if you agree with my groupings. (5/4 and 12/8 still have problems but the others are basically done.)

lasconic commented:
There is nothing to disagree for the grouping. The 5/4 looks good, other possibilities exist and would probably equally good. See for example http://www.musicnotes.com/sheetmusic/mtd.asp?ppn=MN0052300

IsaacWeiss commented on mscore/musescore.cpp:
```diff
@@ -1087,6 +1089,7 @@ MuseScore::MuseScore()
       menuFormat->addSeparator();
 
       menuFormat->addAction(getAction("reset-beammode"));
+      menuFormat->addAction(getAction("reset-groupings"));
```
This might be better placed in the Tools menu after Respell Pitches. Maybe a more concise name, too, like "Regroup Durations" or something.

shoogle replied:
I was considering that too. I ended up putting it with the option to reset the beam mode because they are both related to grouping around a beat, but perhaps you are right and this should be considered a "content change" rather than a "formatting change". I will think about it after GSoC.
